File: searching/binary.js
function binarySearch(_sorted, val){
    var left = 0;
    var right = (_sorted.length-1)
    
    while (left <=  right) {
        let middle = Math.floor((left+right)/2);
        
        if (val > _sorted[middle]){
            left = middle + 1;
        }
        
        else if (val < _sorted[middle]){
            right = middle - 1;
        } 
        
        else {
            return middle;
        }}
        
    return -1;
    
    }

File: searching/binary.py
import math

def binary_search(_sorted, value):
    left = 0
    right = len(_sorted) - 1

    while left <= right:
        middle = math.floor((right + left)/2)
        if value > _sorted[middle]:
            left = middle + 1
        elif value < _sorted[middle]:
            right = middle - 1
        else:
            return _sorted[middle]
        
    return -1
